ILL/IDF/d7_generateIDF.py

```diff
@@ -0,0 +1,120 @@
+       Generated file, PLEASE DO NOT EDIT THIS FILE!
+       This file was automatically generated by mantidgeometry/ILL/IDF/d7_generateIDF.py
+       z axis defines the direction of the beam
+       y axis will be the axis used for rotation
+       coordinate system is right-handed
+       x axis rotation defined by phi 
+       y axis rotation defined by theta 
+       height y direction
+       The instrument operates in two modes
+       - Diffraction:
+          - wavelength 3.1 A, qmin = 0.4, qmax = 4.0
+          - wavelength 4.8 A, qmin = 0.3, qmax = 2.5
+          - wavelength 5.7 A, qmin = 0.2, qmax = 2.1
+       - Spectroscopy time-of-flight 
+          - using Fermi chopper, with the same possible wavelengths
+
+       Source-to-sample distances is 3.6 m 
+       Three detector banks with 44 position-insensitive He3 detectors each
+       Each detector located 1.5 m from the sample
+       covering 2 Theta range from about 10 to 155 degrees
+       Detector tubes have 25 mm in diameter and are 250 mm in height
+       For more information, please visit 
+       https://www.ill.eu/instruments-support/instruments-groups/instruments/d7/characteristics/
+       """
+d7 = MantidGeom(instrumentName, comment=comment, valid_from=validFrom)
+d7.addSnsDefaults(default_view='3D',axis_view_3d='z-')
+d7.addComment("SOURCE")
+d7.addComponentILL("Monochromator", 0., 0., source, "Source")
+d7.addComment("Sample position")
+d7.addComponentILL("sample_position", 0., 0., 0., "SamplePos")
+d7.addComment("Fermi chopper")
+d7.addComponentILL('fermi_chopper', 0.0, 0.0, zChopper, 'ChopperPos')
+d7.addComment("MONITORS")
+d7.addMonitors(names=["monitor1", "monitor2"], distance=[zMon1, zMon2])
+d7.addComment("MONITOR SHAPE")
+d7.addComment("FIXME: Do something real here.")
+d7.addDummyMonitor(0.01, 0.03)
+d7.addComment("MONITOR IDs")
+d7.addMonitorIds([repr(1000), repr(1001)])
+d7.addComment("DETECTORS")
+d7.addComponentILL("detector", 0., 0., 0.)
+detector = d7.makeTypeElement("detector")
+
+# define detector banks
+d7.addDetectorIds("bank2_ids", [1,44,1]) 
+d7.addDetectorIds("bank3_ids", [45,88,1]) 
+d7.addDetectorIds("bank4_ids", [89,132,1]) 
+bank = d7.makeTypeElement("bank")
+bank2 = d7.addComponent("bank", idlist = "bank2_ids", root=detector)
+bank3 = d7.addComponent("bank", idlist = "bank3_ids", root=detector)
+bank4 = d7.addComponent("bank", idlist = "bank4_ids", root=detector)
+
+# define detector banks initial positions
+d7.addLocation(root=bank2, x = 0, y = 0, z = 0, rot_y = repr(-(minTwoTheta + 0.5 * numberDetectors)), name="bank2")
+d7.addLocation(root=bank3, x = 0, y = 0, z = 0, rot_y = repr(-(minTwoTheta + 1.5 * numberDetectors + 2 )), name="bank3")
+d7.addLocation(root=bank4, x = 0, y = 0, z = 0, rot_y = repr(-(maxTwoTheta - 0.5 * numberDetectors)), name="bank4")
+
+bankComponent = d7.addComponent(root=bank, type_name="pixel")
+
+# define detectors and their relative position inside the bank:
```

Comment:
maybe just make clear how you call odd/even, do you start from 0 or 1? is it in terms of detector ID or index?

Reply:
It does merit clarifying, the first detector has an index of 0 and is located at a larger radius, thus it is treated as an 'even' detector. Added comment to the first mention of the even/odd detectors, in line 38, that the indices start from '0' (even). Unless this is not clear the description and calculation can be easily changed.

Reply:
The first detector has index 0, but name pixel_1, that's what creates confusion in defining what is odd/even. Can you put this clarification inside the code as a comment?

Reply:
Added comment to the line setting pixel ids: "pixel indices start from 1, while detector IDs start from 0", and few lines below, after the conditional statement: " odd detectors, according to their detector ID, are closer to the sample". Do you think that is sufficient?

Reply:
If we do this `d7.addDetectorIds("bank2_ids", [1,44,1]) `how do detector IDs start from 0, they start at 1?

Reply:
They indeed did not, which is corrected in the most recent commit, where the indices start from 0.